### ISSUE TITLE
Terms of usage for the photos in westnordost.de

### ISSUE THREAD

**Issue by angoca** [streetcomplete/StreetComplete#3714]
I know that photos are hosted in a website called https://westnordost.de/. But it seems that this hosting site along with its domain name were kindly offered by his owner to be used in this project. However, the main page still point to a personal blog.

I propose that that website has a page about the photos being hosted, with at least the term of usage. When a site hosts photos or any kind of content from other people, you have to define a set of rules, like these other photo sites:

* Mapillary - https://www.mapillary.com/terms
* Karta view - https://kartaview.org/terms
* Wikipedia commons - https://meta.wikimedia.org/wiki/Terms_of_use?rdfrom=commons:Commons:Terms_of_use

The only place in this project that specifies something about the photos is in the privacy policy option, but it is not enough - privacy_html_image_upload2: 

> Photos you attach to a note are uploaded to my server and deleted some days after that note has been resolved. Their meta-data is stripped before upload.

I know some of your hosting terms, because our interactions in other places (my article - https://www.openstreetmap.org/user/AngocA/diary/398514#comment51862, and other places), but they should be more easily to found for general public.

I bother a lot with the usage terms and licenses, because this is the only way to protect our work. That's why OSM is so powerful, because of its ODbL license, similar to the CreativeCommons in Wikipedia, and the license of this application - GPL, and each of these platforms also has usage terms.

Last thing, let's imagine someone publish ilegal photos (restricted area - militar, private photos - pornography) on your site, then you become involved in a bigger problem because you didn't define a terms of usage. What about if a user asks to delete a photo, what is the process? Regarding the deletion from your part:
* what about if the photo link is used as part of the tagging - https://wiki.openstreetmap.org/wiki/Key:image ?
* What about if the note is solved several years after the note has been created? are you going to keep photos for decades?
* What about if you need to delete old photos because a storage issue, and you destroy photos that haven't been used yet to solve notes?

Finally, and because I appreciate a lot this application, which it is an important tool of my mapping work, I do not want to fall in a legal issue, just because it lacks a formal terms of usage.

**Comment by westnordost** [comment]
In my own words: I don't see any missing terms of usage as an issue. 

The user obviously uploaded the photo to help solving an issue in OpenStreetMap. If he did not intend the photo to be used this way, he would not have uploaded it because the context is clear and also mentioned in the privacy statement. In other matters, you can consider any photo uploaded there as **"All rights reserved"** by the uploader of the photo, because that's the default if there is no other statement.
So technically, the most strict understanding is that neither I nor anyone else may use/license/sell/publish these photos anywhere else than for the purpose it was uploaded for and this is fine. Whether they care or not is another matter, those are just random street snapshots after all.

So when a photo turns out to be illegal, the user can just close the note and the picture will be deleted. The photo will stay there until the note it is "attached" to is closed or the allocated space on my webspace runs out (then the oldest "attached" photos will be deleted first). This means that the links to very old notes that have a photo attached will return a 404 Not Found error. Space will definitely not run out for the next couple of years, and if it does, maybe I'll just rent more space. Or, maybe attaching images to notes will be implemented on the OpenStreetMap website by then. That would be nice.

**Comment by angoca** [comment]
In Google maps, as part of the terms, [they say](https://www.google.com/intl/en/help/terms_maps/):

> 2. Prohibited Conduct. Your compliance with this Section 2 is a condition of your license to use Google Maps/Google Earth. When using Google Maps/Google Earth, you may not (or allow those acting on your behalf to):
>
>  d. use Google Maps/Google Earth to create or augment any other mapping-related dataset (including a mapping or navigation dataset, business listings database, mailing list, or telemarketing list) for use in a service that is a substitute for, or a substantially similar service to, Google Maps/Google Earth; or

Also, they explicitly express their managerial status of the [uploaded content](https://support.google.com/contributionpolicy/answer/7422880?hl=ALL).

----

Taking the previous part as reference, am I permitted to use photos published on westnordost.de to create derivative works? For the moment, I haven't seen if I can do it or not, and I am probably violating some un-explicit terms. As you know, I am very interested in solving notes, not create them, and I need to know if the material to solve the notes has a compatible license with OSM license - ODbL. I know you agree with this, but this is not stated anywhere.

Let's go to the details. There are two legal things here we are talking about, but they are completely different:

* License of the photos.
* Terms of the website.

# License of the photos

All photos should have a license, and the person uploading them have to know its rights, even if it is something like he will lose all rights on the photo. For example a text like this: "The photo will be deleted once the note is closed or if the server runs out of space. The sole purpose of the photo is to better describe the note. The photo will be published under [CC0 license](https://creativecommons.org/share-your-work/public-domain/cc0/) (or whatever license you choose). The content of the photo is not responsibility of the website or his owner."

For example, if someone takes a great photo and publish it on your site; then another person download it, and gains something with it, like money by selling it, or obtaining a Pulitzer, then the license will become a problem, and your site could be in the middle of a legal battle.

Remember, public things are not necessary free of use things, and that's where the problem starts.

Also, many sites protect themselves after the European GPDR.

Regarding "All rights reserved", this could include the right to not public publishing it. And the fact that photos are public published in your site, your are going against that right. I know it is crazy, but sometimes the interpretation of the law is crazy.

# Terms of your website

Is your website, and the owner of the domain responsible of the usage and content published there?

Imagine someone stole graphic information (photo of blueprints, medical information), and publish it in your site. Because you do not capture any information of the person who uploaded (at least you said you delete all metadata, and notes could be anonymous), the only responsible about that publication is you.

Also, let's think if your website is selected/used by bad people (hackers), to transfer ilegal photos. And what about if they know that the photo is deleted when the note is closed. They could use this mechanism to anonymize their activities with your site, and the responsible will be you.
Hacker always try to find places or ways where to anonymize their activities, and your site could be a very good option.

Another crazy case: let's suppose someone uploads a photo (associated with a note in a place nobody cares about it, so no-one will close it), and use your site as a storage server. Then, imagine the person links the photo in another website, or any other professional usage. Suddenly you delete that photo because you run out of space, or whichever other reason. If the fact of deleting that photo creates an impact on the other website (because the photo is missing), you are the responsible! Your action could run out of business a company, and they could accuse you, because you didn't have terms of usage.

Regarding what you said, about "The user obviously uploaded the photo to help solving an issue in OpenStreetMap", for me there are not obvious things. After closing more than 5000 notes, I have found information from any kind in them, personal phone number, exact location of someone's house, and even things like police reports. With photos this could be even worse.

I have googled a little bit about the importance of having legal terms (website terms of usage), and I found:

* https://www.legalnature.com/guides/why-your-website-needs-a-strong-terms-of-use-agreement-and-what-to-include
* https://www.godaddy.com/garage/does-your-website-really-need-terms-of-use/

----

Even, OpenStreetMap has two pages, one for each of the elements previously described:

* https://www.openstreetmap.org/copyright
* https://wiki.osmfoundation.org/wiki/Terms_of_Use

Finally, this is just an advise. But if you found that this is bothering you, that I am trying to accuse you, and you think you are currently doing the right thing, please close the note. Thank you.

**Comment by matkoniecz** [comment]
There is bunch of scenarios here where I am quite confused how ToU would help with anything.

> am I permitted to use photos published on westnordost.de to create derivative works? For the moment, I haven't seen if I can do it or no

No, unless it is your photo AND it is not a derivative work of something else within photo (unless freedom of panorama applies). Copyright applies by default.

ToC are not necessary for copyright to apply.

> and I need to know if the material to solve the notes has a compatible license with OSM license - ODbL.

You are free to extract facts from copyrighted images, for example mapping based on info from TV that say specific `natural=tree` was cut down is fine.

Problems start when you copy from maps/databases (or photos of them) but in such case ToC would not change anything.

> For example, if someone takes a great photo and publish it on your site; then another person download it, and gains something with it, like money by selling it, or obtaining a Pulitzer, then the license will become a problem, and your site could be in the middle of a legal battle.

ToC would not change anything here. Someone can be sued without any good reason and operating public service increases risk. But how having ToC would reduce risk of egregiously dumb lawsuits?

> https://www.legalnature.com/guides/why-your-website-needs-a-strong-terms-of-use-agreement-and-what-to-include

Obviously, lawyers would recommend paying to lawyers. I would trust it more if they would bother to provide sources for their claims.

> Imagine someone stole graphic information (photo of blueprints, medical information), and publish it in your site. Because you do not capture any information of the person who uploaded (at least you said you delete all metadata, and notes could be anonymous), the only responsible about that publication is you.
> Also, let's think if your website is selected/used by bad people (hackers), to transfer ilegal photos. And what about if they know that the photo is deleted when the note is closed. They could use this mechanism to anonymize their activities with your site, and the responsible will be you.

"the only responsible about that publication is you"

[citation needed]

Even if that would be true, how ToU would change this?


> Another crazy case: let's suppose someone uploads a photo (associated with a note in a place nobody cares about it, so no-one will close it), and use your site as a storage server. Then, imagine the person links the photo in another website, or any other professional usage. Suddenly you delete that photo because you run out of space, or whichever other reason. If the fact of deleting that photo creates an impact on the other website (because the photo is missing), you are the responsible! Your action could run out of business a company, and they could accuse you, because you didn't have terms of usage.

Anyone can sue for any reason. Also, [citation needed] that it is actually a sound risk. Also, if they run out of business then they would not afford malicious and ridiculous litigation either.

Even if that would be true, how ToU would change this?

> All photos should have a license

Not necessary. In this case default "all rights reserved" applies.

> Regarding "All rights reserved", this could include the right to not public publishing it. And the fact that photos are public published in your site, your are going against that right. I know it is crazy, but sometimes the interpretation of the law is crazy.

If that would be true and valid problem then just having ToU solves nothing.

If implied consent by user taking photos and sharing them for express purpose of public sharing is not enough and not covered by fair use equivalent... Then just ToU somewhere is not going to help and user would need to explicitly accept them.

> The photo will be published under [CC0 license](https://creativecommons.org/share-your-work/public-domain/cc0/) (or whatever license you choose)

Many photos taken cannot be published under open license. Note countries without https://commons.wikimedia.org/wiki/Commons:Freedom_of_panorama for stat.

**Comment by matkoniecz** [comment]
There are multiple parts here. Education about copyright is not relevant at all.

The part that is potentially useful is reducing legal risk but the question is whether in German jurisdiction it is actually noticeable help.

**Comment by angoca** [comment]
Then, if everything is OK as it is now without ToU, then why don't you close this issue?

**Comment by 10992-osm** [comment]
> The part that is potentially useful is reducing legal risk but the question is whether in German jurisdiction it is actually noticeable help.

And the only person who can *really* help with that (beyond what has already been done, I should think) is a German lawyer, not random people on the Internet.

**Comment by mnalis** [comment]
Firstly @angoca I'd like to thank you for putting effort and reporting your concerns :heart:. Even if the decision happens to be that nothing needs to be changed, it's appreciated. 

Also IANAL (so consult relevant lawyer for legal advice); but I also care about the issue, so here are my thoughts:

> * what about if the photo link is used as part of the tagging - https://wiki.openstreetmap.org/wiki/Key:image ?

It should not be, as it is noted in Privacy Policy that the image store on SC servers are ephemeral. 

>     * What about if the note is solved several years after the note has been created? are you going to keep photos for decades?
>     * What about if you need to delete old photos because a storage issue, and you destroy photos that haven't been used yet to solve notes?

Already answered - it might go away. Or you know what? The server might crash, domain might be lost etc. At best, one damaged by it might ask for money payed for that service to be refunded. There is likely no guarantee implied unless you entered into service agreement (and likely payed for it). As it is free of charge, one would probably just have to eat up that loss. If gmail for example classified all your mail as spam in your free-of-charge account and deleted it, do you think likely you'd be able to get money from them, ToS or no ToS?


> Taking the previous part as reference, am I permitted to use photos published on westnordost.de to create derivative works?

No. By default in most any country which has copyright law, all photos are copyrighted by default. Which means, unless you were given other rights, "All rights reserved" condition applies. Which means you may not create derivative works, limited only by "fair use" or similar clauses in copyright law that applies (which might involve international law expert to properly define).

So basically any photo taken with SC must be considered as any other random photo found on the Internet and without specific license attached to it. Which is, you must get permission of copyright owner for any activity needed which is covered by copyright law.

Note however, that looking at the photo, and then creating other work based on the information you gleaned from it, is not "creating derivative work" in any copyright system I've seen. Copying or modifying parts of the photo however is creating derivative work. (Incidentally, so is unauthorized copying parts of Google ToS which you did above; so unless that is considered fair use in relevant jurisdiction, or you've got permission from Google to do so, you've just infringed on Googles' copyright)

(Of course, if it is your photo, and it does not infringe on any other copyrights, you can create derivative works.)

> All photos should have a license, and the person uploading them have to know its rights

Photos _"should have a license"_ only if one has interest in licensing them. That is akin to saying _"All people should have a driving license"_ - only true if all those people want to drive motor vehicle or whatever. If they're content to be pedestrians, they don't need driving license. Same with photos - unless you intend to change default "All rights reserved" policy, there is no need to license them.

While I agree it would be marginally more useful if SC photos in Notes were licensed under more permissive license, do note that it could be more problematic. Firstly, you may likely not require photographers to relinquish some of their rights just by using the app - there needs to be explicit _"I agree"_ form for it (and associated infrastructure, keeping track of what users allowed that and when, from which IP they came, GDPR rules for using databases of personally identifiable information, etc.) - that is a lot of work and obligations for relatively small improvement (as photos in SC are not intended to be reusable works - they are short lived and of deliberately reduced quality - so when I want to keep some SC photos, I take both in SC and directly in Camera app.)
 
Secondly, as @matkoniecz notes, pictures might have copyright of more people than just the photographer who uploaded SC Notes (e.g. architect that built building depicted in picture etc).

So probably your best course of action if you want to create derivative work of someone else's picture is SC Note, is to contact  them directly and obtain needed permissions.

> For example, if someone takes a great photo and publish it on your site; then another person download it, and gains something with it, like money by selling it, or obtaining a Pulitzer, then the license will become a problem, and your site could be in the middle of a legal battle.

Sure, but it will be in middle of legal battle anyway, no matter on the amount of ToS it had. All kind of crazy things are possible, it's just that many are not probable at all. For example, Google could sue Microsoft for your unauthorized reproduction of part of Google ToS of GitHub website.  Or for example you could decide to sue me for reproducing part of your text in my reply. I wouldn't lose much sleep over either possibility, though.  YMMV. 

> After closing more than 5000 notes, I have found information from any kind in them, personal phone number, exact location of someone's house, and even things like police reports. With photos this could be even worse.

Sure, errors happen. Photos however will get deleted after few days when note is closed, while notes with personal textual information (even when closed) will remain [available likely forever](https://planet.openstreetmap.org/notes/). However do note that this is Internet - once the cat is out of the bag, it is likely never going back in. If you're celebrity and you've published your sex photos by mistake on SC, I guess you'll better learn to live with it.

> Regarding "All rights reserved", this could include the right to not public publishing it.

It does. However, explicit act of copyright owner uploading his work to public place for public use waives that right, AFAIK. Otherwise, you could go on public square and sing, and then sue everyone who was on that square to pay for your perfomance because they enjoyed your singing. (You could try to sue them of course, but only result would be you losing money.)

> Also, let's think if your website is selected/used by bad people (hackers), to transfer ilegal photos

Sure, it's possible. I guess when site owner becomes aware of, he would likely shut it down, and that will be the end of people being able to upload photos with SC. It would be quite sad, but hey.
Do note: when someone breaks into your home or your machine or steals your car and commits a crime with it, that does not make you liable for their crimes. It _would_ get you involved with authorities until you explain your innocence. Note that ToS or no ToS wouldn't prevent that (the same as a post in on your car saying "you are not allowed to steal this car and commit a crime with it" would not stop police coming to your house after finding  the car with your license plate was involved in bank robbery.  It might (or not) make your explaining somewhat easier, though.

> Finally, this is just an advise. But if you found that this is bothering you, that I am trying to accuse you, and you think you are currently doing the right thing, please close this issue. Thank you.

Also note, the photographer is required by law not to take (much less publish) unlawful pictures. They also must have accepted OpenStreetMap ToS when they created account there (or they wouldn't be able to login with SC, nor upload photos as consequence).

So, do you think there remains issues not covered by laws and OSM ToS, that needs to be spelled out in SC ToS?

**Comment by westnordost** [comment]
> Another crazy case: let's suppose someone uploads a photo (associated with a note in a place nobody cares about it, so no-one will close it), and use your site as a storage server. Then, imagine the person links the photo in another website, or any other professional usage. Suddenly you delete that photo because you run out of space, or whichever other reason. If the fact of deleting that photo creates an impact on the other website (because the photo is missing), you are the responsible! Your action could run out of business a company, and they could accuse you, because you didn't have terms of usage.

Oof, really? 🤣 I find it really funny that you think it works that way.

**Comment by westnordost** [comment]
Anyway, I didn't close the issue because I didn't want to come across as rude and give you a chance to reply. I also find the other concerns/examples you raise to be not convincing that they are correct / do apply / realistic. And I think I better close this issue now before every contributor to this app tries to refute every one of your scenarios (was too late for @mnalis ).

Just in short to wrap this up:
1. The uploader implicitly gives permission to host his picture by the deliberate act of uploading it, the same way you give permission to read your post when you comment here or anywhere else. No license is required. In doubt, the photo may not be reused for any other purpose as for the intent the user uploaded it for. 
2. Misuse / Vandalism is very unlikely because there are far easier ways to share files (illegally) on the internet. As far as I know, it hasn't occured yet. If it occurs, it can easily be dealt with by just deleting/closing the note.
3. The nature of the photos uploaded there is volatile because they are deleted when the note is closed and this is deliberate. Thus, linking to them from the outside directly (from website or from `Key:image` is not going to make sense
4. As basically any webspace, there are access logs that are kept for a certain time automatically by the webspace hoster, so in case of illegal activity and an investigation I'd think this would be handed over. Posting things to some website is not anonymous. The hoster is in Germany, not in Dubai.

**In short: I am not concerned.**
